pkg/prometheus/promcfg.go

```diff
@@ -2273,6 +2273,48 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "scheme", Value: strings.ToLower(*sc.Spec.Scheme)})
 	}
 
+	if sc.Spec.ProxyUrl != nil {
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "proxy_url",
+			Value: *sc.Spec.ProxyUrl,
+		})
+	}
+
+	if sc.Spec.NoProxy != nil {
+		cfg = append(cfg, yaml.MapItem{
```

Comment:
we need a version check (e.g. Prometheus >= 2.43.0).
```suggestion
		cfg = cg.WithMinimumVersion("2.43.0").AppendMapItem(cfg, yaml.MapItem{
```